### PR TITLE
Add robot keywords for generating and trusting a certificate authority

### LINKED ISSUE #4582: Need to establish signed public CA testing procedure
User statement: As a customer of VIC, I might use public signed CA certs and want to be assured that the engine, harbor, and admiral have been tested to work properly with them

Details: 
letsencrypt.org
https://github.com/andrewtchin/ca-test
https://github.com/cloudflare/cfssl

Acceptance criteria:
A keyword or set of keywords that provides public signed certs to be used in harbor/admiral/engine installs.

### PR DESCRIPTION
**Pull request by andrewtchin**
Fixes #4582 

Adds keywords to use certificates, updates some vic-machine TLS tests to use them.

### CODE REVIEW

File: tests/resources/Cert-Util.robot [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    Log  ${out}
+
+
+
+Generate Server Key And CSR
+    # Generates key and CSR (private/DOMAIN.key.pem, csr/DOMAIN.csr.pem) in OUT_DIR
+    [Arguments]  ${CN}=%{DOMAIN}  ${OUT_DIR}=/root/ca
+    Log To Console  Generating Server Key And CSR
+    ${rc}  ${out}=  Run And Return Rc And Output  generate-server-key-csr.sh -d ${OUT_DIR} -n ${CN}
+    Log  ${out}
```
Comment (mhagen-vmware) [inline tests/resources/Cert-Util.robot]: No need for RC if you don't look at it.

File: tests/resources/Cert-Util.robot [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+
+
+Sign Server CSR
+    # Generates certificate signed by CA (certs/DOMAIN.cert.pem) in OUT_DIR
+    [Arguments]  ${CA_NAME}=STARK_ENTERPRISES_ROOT_CA  ${CN}=%{DOMAIN}  ${OUT_DIR}=/root/ca
+    Log To Console  Signing Server CSR
+    ${rc}  ${out}=  Run And Return Rc And Output  sign-csr.sh -c ${CA_NAME} -d ${OUT_DIR} -n ${CN}
```
Comment (mhagen-vmware) [inline tests/resources/Cert-Util.robot]: All of these can be simplified to just Run and log the output

**Comment by mhagen-vmware** [comment]
You don't need to implement 19-2, that is scope creep.  Just fix the other things and we can merge this.  Thanks!

File: tests/resources/Cert-Util.robot [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+*** Settings ***
+Documentation  This resource contains keywords related to creating and using certificates. Requires scripts in infra/integration-image/scripts be available in PATH
+
+*** Keywords ***
+Generate Certificate Authority
+    #  Generates CA (private/ca.key.pem, certs/ca.cert.pem, certs/STARK_ENTERPRISES_ROOT_CA.crt) in OUT_DIR
+    [Arguments]  ${CA_NAME}=STARK_ENTERPRISES_ROOT_CA  ${OUT_DIR}=/root/ca
+    Log To Console  Generating Certificate Authority
+    ${rc}  ${out}=  Run And Return Rc And Output  generate-ca.sh -c ${CA_NAME} -d ${OUT_DIR}
```
Comment (andrewtchin) [inline tests/resources/Cert-Util.robot]: theyre in `/opt/vmware/scripts` and that dir is added to PATH

File: tests/resources/Cert-Util.robot [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    Log  ${out}
+    Should Be Equal As Integers  ${rc}  0
+
+
+
+Generate Wildcard Server Certificate
+    # Generates key and signs with CA for *.DOMAIN (csr/*.DOMAIN.csr.pem,
+    # private/*.DOMAIN.key.pem, certs/*.DOMAIN.cert.pem) in OUT_DIR
+    [Arguments]  ${DOMAIN}=%{DOMAIN}  ${OUT_DIR}=/root/ca  ${CA_NAME}=STARK_ENTERPRISES_ROOT_CA
+    Log To Console  Generating Wildcard erver Certificate
```
Comment (andrewtchin) [inline tests/resources/Cert-Util.robot]: fixed

File: tests/resources/Cert-Util.robot (hunk 1 above)
Comment (andrewtchin) [inline tests/resources/Cert-Util.robot]: fixed

File: tests/resources/Cert-Util.robot (hunk 2 above)
Comment (andrewtchin) [inline tests/resources/Cert-Util.robot]: fixed

File: tests/resources/Cert-Util.robot [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+Generate Certificate Authority
+    #  Generates CA (private/ca.key.pem, certs/ca.cert.pem, certs/STARK_ENTERPRISES_ROOT_CA.crt) in OUT_DIR
+    [Arguments]  ${CA_NAME}=STARK_ENTERPRISES_ROOT_CA  ${OUT_DIR}=/root/ca
+    Log To Console  Generating Certificate Authority
+    ${rc}  ${out}=  Run And Return Rc And Output  generate-ca.sh -c ${CA_NAME} -d ${OUT_DIR}
+    Log  ${out}
+    Should Be Equal As Integers  ${rc}  0
+
+
+
```
Comment (andrewtchin) [inline tests/resources/Cert-Util.robot]: fixed

**Pull request merged**